modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManagerSelectBox.vue

```diff
@@ -56,6 +60,18 @@ export default {
   },
   emits: [ 'select-all', 'clear-select', 'set-all-pieces-selection' ],
   computed: {
+    selectionBeyondCurrentPage() {
+      return !!this.checkedIds.filter(x => !this.currentItems.map(y => y._id).includes(x)).length;
+    },
+    showSelectAll() {
+      if (
+        this.selectionBeyondCurrentPage ||
```

Comment:
Here you can simply relace `this.selectionBeyondCurrentPage` by `this.allPiecesSelection.isSelected` and remove `selectionBeyondCurrentPage` to avoid more computation.